### PR TITLE
Improve the code

### PR DESCRIPTION
**Pull request by lexign**
1. :art: set SequelizeAdapter extends default Adapter;
2. :hammer: add constructor method for the class;
3. :wrench: add options for new Sequelize.

### CODE REVIEW

File: test/adapter.test.ts [hunk 1]
```diff
@@ -53,7 +53,7 @@ test('TestAdapter', async () => {
     // Now the DB has policy, so we can provide a normal use case.
     // Create an adapter and an enforcer.
     // newEnforcer() will load the policy automatically.
-    a = await SequelizeAdapter.newAdapter('mysql://root:@localhost:3306/');
+    a = await SequelizeAdapter.newAdapter('mysql://root:@localhost:3306/casbin', true);
```
Comment (hsluoyz) [inline test/adapter.test.ts]: Test coverage downgrade is OK, no need for this change.

File: src/adapter.ts [hunk 2]
```diff
@@ -105,7 +112,7 @@ export class SequelizeAdapter {
     /**
      * loadPolicy loads all policy rules from the storage.
      */
-    public async loadPolicy(model: Model) {
```
Comment (hsluoyz) [inline src/adapter.ts]: The things like ``Promise<void>`` is not necessary. It will be automatically returned.

File: src/adapter.ts [hunk 3]
```diff
@@ -142,7 +149,7 @@ export class SequelizeAdapter {
     /**
      * savePolicy saves all policy rules to the storage.
      */
-    public async savePolicy(model: Model) {
+    public async savePolicy(model: Model): Promise<boolean> {
```
Comment (hsluoyz) [inline src/adapter.ts]: The things like ``Promise<void>`` is not necessary. It will be automatically returned.

File: src/adapter.ts [hunk 4]
```diff
@@ -163,26 +170,28 @@ export class SequelizeAdapter {
...
         }
+
+        return true;
     }
 
     /**
      * addPolicy adds a policy rule to the storage.
      */
-    public addPolicy(sec: string, ptype: string, rule: string[]) {
+    public async addPolicy(sec: string, ptype: string, rule: string[]): Promise<void> {
```
Comment (hsluoyz) [inline src/adapter.ts]: The things like ``Promise<void>`` is not necessary. It will be automatically returned.

File: src/adapter.ts [hunk 5]
```diff
@@ -163,26 +170,28 @@ export class SequelizeAdapter {
...
         throw new Error('not implemented');
     }
 
     /**
      * removePolicy removes a policy rule from the storage.
      */
-    public removePolicy(sec: string, ptype: string, rule: string[]) {
+    public async removePolicy(sec: string, ptype: string, rule: string[]): Promise<void> {
```
Comment (hsluoyz) [inline src/adapter.ts]: The things like ``Promise<void>`` is not necessary. It will be automatically returned.

File: src/adapter.ts [hunk 6]
```diff
@@ -163,26 +170,28 @@ export class SequelizeAdapter {
...
         throw new Error('not implemented');
     }
 
     /**
      * removeFilteredPolicy removes policy rules that match the filter from the storage.
      */
-    public removeFilteredPolicy(sec: string, ptype: string, rule: string[]) {
+    public async removeFilteredPolicy(sec: string, ptype: string, fieldIndex: number, ...fieldValues: string[]): Promise<void> {
```
Comment (hsluoyz) [inline src/adapter.ts]: The things like ``Promise<void>`` is not necessary. It will be automatically returned.

File: src/adapter.ts [hunk 7]
```diff
@@ -32,18 +32,22 @@ export class SequelizeAdapter {
      * If dbSpecified == true, you need to make sure the DB in connStr exists.
      * If dbSpecified == false, the adapter will automatically create a DB named 'casbin'.
      */
```
Comment (hsluoyz) [inline src/adapter.ts]: The comment is for newAdapter

File: src/adapter.ts [hunk 8]
```diff
@@ -32,18 +32,22 @@ export class SequelizeAdapter {
...
-    public static async newAdapter(connStr: string, dbSpecified: boolean = false) {
-        const a = new SequelizeAdapter();
-        a.connStr = connStr;
-        a.dbSpecified = dbSpecified;
+    constructor(connStr: string, dbSpecified: boolean = false) {
```
Comment (hsluoyz) [inline src/adapter.ts]: No need for default value.

File: src/adapter.ts [hunk 9]
```diff
@@ -52,12 +56,15 @@ export class SequelizeAdapter {
     }
 
     private async open() {
-        if (this.dbSpecified) {
-            this.sequelize = new Sequelize(this.connStr);
+        const {connStr, dbSpecified} = this;
```
Comment (hsluoyz) [inline src/adapter.ts]: Why not just use ``this.connStr``

**Review by hsluoyz — COMMENTED** [review]
Please fix.

File: src/adapter.ts (hunk 3 above)
Comment (lexign) [reply src/adapter.ts]: These methods extend default Adapter interface, must strict at here.

File: src/adapter.ts (hunk 8 above)
Comment (lexign) [reply src/adapter.ts]: I think this is not a problem.

File: test/adapter.test.ts (hunk 1 above)
Comment (lexign) [reply test/adapter.test.ts]: Ok, I'll remove it.

File: src/adapter.ts (hunk 9 above)
Comment (lexign) [reply src/adapter.ts]: If the this.xx variable use twice or more, maybe like this will better.

File: src/adapter.ts (hunk 8 above)
Comment (hsluoyz) [reply src/adapter.ts]: Just fix it

File: src/adapter.ts (hunk 9 above)
Comment (hsluoyz) [reply src/adapter.ts]: In my understanding (at least for other languages like C), ``this`` is just a pointer. And ``this.xx`` doesn't cost much time. If you have evidence that Node.js costs a lot of time in ``this.xx``, please show me.

Comment (lexign) [reply src/adapter.ts]: Just save coding time and short variable name.

Comment (lexign) [reply src/adapter.ts]: You said fix means remove extend?

**Pull request closed without merging**